mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -607,7 +611,8 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
       intermediateRDDStorageLevel: StorageLevel = StorageLevel.MEMORY_AND_DISK,
       finalRDDStorageLevel: StorageLevel = StorageLevel.MEMORY_AND_DISK,
       checkpointInterval: Int = 10,
-      seed: Long = 0L)(
+      seed: Long = 0L,
+                           instr: Option[Instrumentation[ALS]] = None)(
```

Comment:
Modifying training is to get the number of features without collect. I agree with you that we should not change train. But I can't get number of features without changing it.